ReactiveCocoa/UIKit/UIControl.swift

```diff
@@ -10,50 +10,61 @@ import ReactiveSwift
 import UIKit
 import enum Result.NoError
 
-private class UnsafeControlReceiver<Control: UIControl>: NSObject {
-	private let observer: Observer<Control, NoError>
+extension Reactivity where Reactant: UIControl {
+	internal var associatedAction: Atomic<(action: CocoaAction<Reactant>, controlEvents: UIControlEvents, disposable: Disposable)?> {
+		return associatedObject(reactant,
+		                        key: &associatedActionKey,
+		                        initial: { _ in Atomic(nil) })
+	}
 
-	fileprivate init(observer: Observer<Control, NoError>) {
-		self.observer = observer
+	public var action: CocoaAction<Reactant>? {
+		return associatedAction.value?.action
 	}
 
-	@objc fileprivate func sendNext(_ receiver: Any?) {
-		observer.send(value: receiver as! Control)
+	public var controlEventsForAction: UIControlEvents? {
+		return associatedAction.value?.controlEvents
 	}
-}
 
-extension Reactivity where Reactant: UIControl {
-	public func trigger(for events: UIControlEvents) -> Signal<Reactant, NoError> {
+	public func setAction(_ action: CocoaAction<Reactant>?, for controlEvents: UIControlEvents) {
```

Comment:
Did we agree to go with `ReactiveExtensionProvider`/`Base` instead of `Reactivity`/`Reactant`?